### DISCUSSION TITLE
Soroban Tool – Deploy Contracts in Browser

### DISCUSSION THREAD

**Discussion by wildework** [stellar/sorobanathon#159] [General]
I'm building Soroban tools for the web and this is my first useful product. By using it you can essentially deploy Soroban contracts only using your mouse.

The SDK I built to achieve this will enable a lot more features down the line, but I wanted to share my first milestone. It's not yet ready for upload to sorobandev dot com but I have recorded a video of me using the tool to deploy a contract on local net.

https://www.youtube.com/watch?v=jF3-Y1xGf0E

Here's a screenshot of how it looks once you've uploaded your contract.

<img width="1127" alt="Screenshot 2022-12-15 at 23 26 44" src="https://user-images.githubusercontent.com/50962453/207970759-d55c1885-1e38-40b6-940e-2112bf57a4c0.png">

To achieve this, I started building a lightweight Soroban SDK for JavaScript. To install a contract, the code would look like this.

```javascript
const network = Network.local;
const code = new Code(state.contractCode);
const transaction = await new Transaction()
  .sourceAccount(account)
  .operation(
    new Operation.InstallContractCode()
      .code(code)
  )
  .serializedFor(network);
  
await transaction.signedBy(account);

const response = await network.sendTransaction(transaction.finalized());
```

**Comment by achallagundla** [comment]
Heya @wildework ,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1000 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.
